### PR TITLE
Added an example for configuring Elytron OIDC Client with Additional Scope Values

### PR DESCRIPTION
**Pull request by PrarthonaPaul**
This example is based on a new feature on WildFLY and WildFly-Elytron.

### CODE REVIEW

File: elytron-oidc-client-scope/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+We will be making use of two projects inside the repo: `elytron-oidc-client-scope` and `simple-webapp-oidc`. 
+
+In order to authenticate a user using OpenID connect, we need to configure the server to use OpenID attributes. This can be done using one of the two following configurations: 
+
+## Deployment Configuration
+
+The configuration required to secure an application with OpenID Connect can be specified in the deployment.
+
+The first step is to create an `oidc.json` configuration file in the `WEB-INF` directory of the `simple-webapp-oidc` project. The following is the contents of the `oidc.json` file used in this example:
```
Comment (fjuma) [inline elytron-oidc-client-scope/README.md:43]: The simple-webapp-oidc example already includes an oidc.json file. So would be good to mention here that the file will need to be updated and point out the scope configuration that is being added.

File: elytron-oidc-client-scope/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    "scope" : "profile email phone address offline_access",
+    "principal-attribute" : "preferred_username",
+    "ssl-required" : "EXTERNAL"
+}
+```
+
+Note that we have specified the scope values inside the `oidc.json` file. The next step is to set the auth-method to OIDC in the application’s `web.xml` file by adding the following command: 
+```
+<login-config>
+    <auth-method>OIDC</auth-method>
```
Comment (fjuma) [inline elytron-oidc-client-scope/README.md:58]: The example being used should already have this configuration so it would be good to point that out.

File: elytron-oidc-client-scope/README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+Now, let’s try accessing our application using http://localhost:8090/simple-webapp-oidc/.
+
+Click on "Access Secured Servlet".
+
+Now, you’ll be redirected to Keycloak to log in. If you click on the url, you will see the scope values specified in the `redirect-uri` field with the different scope values separated by a `+`. You will also notice that a new scope value, `openid`, has been added. This indicates that we are going to be using OpenID Connect to authenticate the user. 
+
+Log in with `myuser` and the password that you set when configuring Keycloak.
+
+Next, you’ll be redirected back to our application and you should see the "Secured Servlet" page. That means that we’ve been able to successfully log in to our application using the Keycloak OpenID provider!
```
Comment (fjuma) [inline elytron-oidc-client-scope/README.md:103]: Might be good if there was a way to check that the scope value was actually used, e.g., can we check the token to see if it has a specific claim?

Take a look at the oidc-with-bearer example to see how we can obtain the OidcSecurityContext:

https://github.com/wildfly-security-incubator/elytron-examples/blob/main/oidc-with-bearer/app/src/main/java/org/wildfly/security/examples/app/Controller.java#L39

The OidcSecurityContext gives you access to both the access token and the identity token.

File: elytron-oidc-client-scope/simple-webapp-oidc/pom.xml [hunk 4]
```diff
@@ -0,0 +1,70 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (fjuma) [inline elytron-oidc-client-scope/simple-webapp-oidc/pom.xml:1]: I don't think we need a separate subdirectory for simple-webapp-oidc. The code could just be part of the main directory like the structure for the simple-webapp-oidc example.

File: elytron-oidc-client-scope/simple-webapp-oidc/src/main/java/org/wildfly/security/examples/SecuredServlet.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+/**
+ * A simple secured HTTP servlet.
+ *
+ * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
+ */
+@WebServlet("/secured")
+public class SecuredServlet extends HttpServlet {
+    private static final String AUTHZ_HEADER = "Authorization";
+    private static final String ENDPOINT_URL = "http://localhost:8090/service/";
```
Comment (fjuma) [inline elytron-oidc-client-scope/simple-webapp-oidc/src/main/java/org/wildfly/security/examples/SecuredServlet.java:43]: Are the above variables needed?

File: elytron-oidc-client-scope/configure-elytron.cli [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# add the elytron-pidc-client subsystem onto your configuration and a secure-deployment resource.
```
Comment (fjuma) [inline elytron-oidc-client-scope/configure-elytron.cli:1]: s/pidc/oidc

Comment (fjuma) [reply elytron-oidc-client-scope/configure-elytron.cli:1]: maybe could reword this to configure a secure-deployment that specifies the scope attribute

File: elytron-oidc-client-scope/README.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+Click on "Access Secured Servlet".
+
+Now, you’ll be redirected to Keycloak's login page. If you click on the url on the search bar, you will see the scope values specified in the `redirect-uri` field with the different scope values separated by a `+`. You will also notice that a new scope value: `openid`. This indicates that we are going to be using OpenID Connect to authenticate the user. 
+
+Log in with `Alice` and the password that you set when configuring Keycloak.
+
+Next, you’ll be redirected back to our application and you should see the "Secured Servlet" page. That means that we were able to successfully log in to our application using the Keycloak OpenID provider!
+
+This page will display the Current Principal, which is Alice and a list of claim values obtained using the scope values you configured.
```
Comment (fjuma) [inline elytron-oidc-client-scope/README.md:116]: I think it would be good to show what the output will look like here.

File: elytron-oidc-client-scope/charts/helm-2.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+build:
```
Comment (fjuma) [inline elytron-oidc-client-scope/charts/helm-2.yaml:1]: I think we should rename this to helm.yaml since this is the main approach that people should now be using to use the feature.

File: elytron-oidc-client-scope/pom.xml [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+    <version>2.0.0.Alpha1-SNAPSHOT</version>
+    <name>simple-webapp Maven Webapp</name>
+    <url>http://maven.apache.org</url>
+
+    <properties>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+        <version.maven.war.plugin>3.3.2</version.maven.war.plugin>
+        <version.wildfly.maven.plugin>5.0.0.Beta5</version.wildfly.maven.plugin>
+        <version.wildfly>32.0.0.Beta1</version.wildfly>
```
Comment (fjuma) [inline elytron-oidc-client-scope/pom.xml:32]: This can now be updated.

File: elytron-oidc-client-scope/pom.xml [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+        <version.wildfly.cloud.galleon.pack>7.0.0.Beta2</version.wildfly.cloud.galleon.pack>
+        <version.wildfly.security>2.4.0.CR1</version.wildfly.security>
```
Comment (fjuma) [inline elytron-oidc-client-scope/pom.xml:34]: Same here.

File: elytron-oidc-client-scope/charts/helm-2.yaml (hunk 8 above)
Comment (PrarthonaPaul) [reply elytron-oidc-client-scope/charts/helm-2.yaml]: fixed! 
I have renamed the other one to helm-in-progress-feature.yaml for more clarity.

File: elytron-oidc-client-scope/pom.xml (hunk 9 above)
Comment (PrarthonaPaul) [reply elytron-oidc-client-scope/pom.xml]: fixed!

File: elytron-oidc-client-scope/pom.xml (hunk 10 above)
Comment (PrarthonaPaul) [reply elytron-oidc-client-scope/pom.xml]: fixed!

File: elytron-oidc-client-scope/README.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
+## Configuring Elytron OIDC Client with Additional Scope Values 
+
+The [ Securing WildFly Apps with OpenID Connect ](https://wildfly-security.github.io/wildfly-elytron/blog/securing-wildfly-apps-openid-connect/) example demonstrates how to use OpenID Connect to perform authentication on a web app deployed on WildFly. 
+
+[ OpenID Connect ](https://openid.net/developers/how-connect-works/) is an identity layer on top of the OAuth 2.0 protocol that makes it possible for a client to verify a user’s identity based on authentication that’s performed by an OpenID provider. The [ OpenID Connect specifications ](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) indicate that there are other scope values which may be included in the authentication request to ask permission to additional and specific resources.The purpose of this example is to demonstrate how to configure additional scope values to request access to additional fields.
```
Comment (fjuma) [inline elytron-oidc-client-scope/README.md:5]: resources.The 

^ missing space before The

File: elytron-oidc-client-scope/README.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+Next, you’ll be redirected back to our application and you should see the "Secured Servlet" page. That means that we were able to successfully log in to our application using the Keycloak OpenID provider!
+
+This page will display the Current Principal,  and a list of claim values obtained using the scope values you configured. This is what it will look like: 
+```
+Secured Servlet
+Current Principal 'fc27fd69-3995-482c-9fad-0d45b838a804'
+
+Claims received using additional scope values:
+
+Using the "profile" scope, we got User's name: Alice Smith
```
Comment (fjuma) [inline elytron-oidc-client-scope/README.md:121]: The text here should be updated.

File: elytron-oidc-client-scope/src/main/java/org/wildfly/security/examples/SecuredServlet.java [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+    private OidcSecurityContext getOidcSecurityContext(HttpServletRequest req) {
+        return (OidcSecurityContext) req.getAttribute(OidcSecurityContext.class.getName());
+    }
+
+    private void displayScopeClaims(String scope, List<String> message, OidcSecurityContext oidcSecurityContext) {
+        if (scope == null){
+            return;
+        }
+        if (scope.contains("profile")){
+            message.add("By configuring the \"profile\" scope, the \"given_name\" and \"family_name\" claims are present in the access code and have values : " + oidcSecurityContext.getToken().getClaimValueAsString("given_name") + " and "
```
Comment (fjuma) [inline elytron-oidc-client-scope/src/main/java/org/wildfly/security/examples/SecuredServlet.java:91]: s/code/token

Comment (PrarthonaPaul) [reply elytron-oidc-client-scope/src/main/java/org/wildfly/security/examples/SecuredServlet.java]: Fixed till here. 
Thanks @fjuma !

File: elytron-oidc-client-scope/charts/helm-in-progress-feature.yaml [hunk 14]
```diff
@@ -0,0 +1,12 @@
+image:
```
Comment (fjuma) [inline elytron-oidc-client-scope/charts/helm-in-progress-feature.yaml:1]: This isn't needed anymore, right?

Comment (PrarthonaPaul) [reply elytron-oidc-client-scope/charts/helm-in-progress-feature.yaml:1]: This is there in case someone would still like to use the in-progress guide.

Comment (fjuma) [reply elytron-oidc-client-scope/charts/helm-in-progress-feature.yaml:1]: Since that example was provided from your fork and your fork was linked from the guide, I don't think we should add it here (since the correct approach now is to make use of the preview stability level).

Comment (PrarthonaPaul) [reply elytron-oidc-client-scope/charts/helm-in-progress-feature.yaml]: Ohh, I see what you mean. 
Yes, that makes sense. I have added a second commit to drop that file and once this branch gets merged, I will drop that commit on my fork to bring this file back. 
Thanks!

**Review by fjuma — APPROVED** [review]
Thanks @PrarthonaPaul!
